app/renderer/actions/Session.js

```diff
@@ -538,17 +540,28 @@ export function newSession (caps, attachSessId = null) {
       } catch (ign) {}
     }
 
+
+    const mjpegScreenshotUrl = desiredCapabilities[`appium:${MJPEG_CAP}`] ||
```

Comment:
this feels a bit hacky to me that we need to fetch the value from capabilities. I would rather move this functionality to a separate inspector setting, so the mjpeg server url could be provided separately. Also, do we want to fallback to vanilla screenshots if the url is not accessible?

Reply:
> this feels a bit hacky to me that we need to fetch the value from capabilities

i thought about this, but it seemed straightforward to just use the existing capability feature. if we create a separate control, then the user would have to put it in both places (or we'd have to auto-use the capability under the hood, which seems dangerous). but i am open to it if you think this way will really confuse users.

> Also, do we want to fallback to vanilla screenshots if the url is not accessible?

i don't think so. if the user provides a cap with incorrect details, better that they see that something is broken?

another (more difficult) option would be to bundle an mjpeg client, which would give us finer grained control and be able to tell if we're actually getting any image data, rather than relying on chrome to show a broken image.

but i figured this was a good start.

Reply:
what will be shown in case we don't get anything from the mjpeg server?

Reply:
hmm yeah, right now it's just a broken image. not sure if there's a way in javascript to do some kind of detection to see whether there is actually data there. maybe an API to convert to base64 and see if it's empty? if so we could show a placeholder or a spinner.

Reply:
ok I now have an interval-based checking function that will detect if the image stream stops working and will show the spinner until it starts working again (i was able to see this in action and it works pretty well)